### ISSUE TITLE
[RFE] Thread sorting order possible improvement

### ISSUE THREAD

**Issue by hexa00** [djcb/mu#806]
Hi,
   I'm confused about the threaded sorting order,

   I expected it to sort such that :

The root level threads are sorted as date descending based on the deepest level message,
but have the messages inside these threads sorted as date ascending.

And if you have 1 thread with only 1 subject it works like that so you get

01/01/01  Mail
02/01/01      -> Re: Mail
03/01/01         -> Re: mail

But if you get multiple sub subjects like it often happens to be following a patches list you get :

example:

01/01/01 [PATCH 0/3] My patch series
   01/01/01 [PATCH 2/3] My patch 2
       02/01/01 RE: [PATCH 2/3] My patch 2
   01/01/01 [PATCH 1/3] My patch 1
   01/01/01 [PATCH 3/3] My patch 3

Rather than :

01/01/01 [PATCH 0/3] My patch series
   01/01/01 [PATCH 1/3] My patch 1
   01/01/01 [PATCH 2/3] My patch 2
      02/01/01 RE: [PATCH 2/3] My patch 2
   01/01/01 [PATCH 3/3] My patch 3

I get the impression that the sorting method is applied recursively such that each thread is considered as it's own root-level thread rather than having this algorithm applied only to the real root level threads.

Maybe it's just me being used to thunderbird, but I find it's very hard to follow patches like that with their order kinda messed up, when I search for a patch in a thread it's very hard to see properly where as it was clear when their order was preserved.

Maybe it can be a customization ?...

Note this was posted on the mailling list, adding it here so that it's not forgotten as suggested.

**Comment by djcb** [comment]
We have better thread-sorting in 1.3.x, so closing this at last!
